src/vendor/utils.js

```diff
@@ -71,41 +113,59 @@ function isInteger(n) {
   return Number.isInteger(n);
 }
 
-function isObject(val) {
-  if (missing(val) || isArray(val)) {
-    return false;
-  }
-
-  return typeof val === 'function' || typeof val === 'object';
-}
+const OBJECT_CONSTRUCTOR = {}.constructor;
 
+/*
+ * objects that are best serialized as JSON objects
```

Comment:
I am at a loss to describe it more; between the English saying it is a JSON object, and the code saying that data is any object that is a direct instance of `Object`, and the matching test suite with a number of examples.

```
  it('isData', () => {
    expect(isData({})).toBe(true);
    expect(isData([])).toBe(false);
    expect(isData(chainFrom([]))).toBe(false);
    expect(isData(() => 0)).toBe(false);
    expect(isData(GMTDate.now())).toBe(false);
    expect(isData(new Duration())).toBe(false);
    expect(isData('')).toBe(false);
    expect(isData('test')).toBe(false);
    expect(isData(undefined)).toBe(false);
    expect(isData(null)).toBe(false);
  });
```
I will make a `Data` class; even if not used, it will serve as a conceptual placeholder.

Reply:
A class might make a good illustration; a couple of extra cases in the test suite containing numeric and string literals, as well as non-empty lists and tuples/objects, would probably be sufficient.  The current test cases look a bit like "this is a list of the things that are not Data; everything else is."  If that's the intent, best to make it explicit.

Reply:
Done. I moved `isData` to `Data.js`, which has some other data manipulation functions, along with a new `Data` class:

https://github.com/mozilla-frontend-infra/firefox-health-dashboard/pull/358/files#diff-795f3e2d7bcca65d54e7515cb23c2163R25

I just learned that [`{[name]: value}` is legitimate syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Computed_object_property_names_and_destructuring)